### PR TITLE
8296157: Rename FileMapInfo::space_at() to region_at()

### PR DESCRIPTION
**Pull request by matias9927**
Most of the CDS code refers to a memory blocks inside a CDS file as "regions", represented by the type FileMapRegion. The method FileRegion* space_at() has been renamed to region_at() and all instances of its use have been replaced. Verified with tier 1-4 tests.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296157](https://bugs.openjdk.org/browse/JDK-8296157): Rename FileMapInfo::space_at() to region_at()

### Reviewers
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [272861ed](https://git.openjdk.org/jdk/pull/10937/files/272861edb1ca3559c5dc4e515fbf81341df8add7)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10937/head:pull/10937` \
`$ git checkout pull/10937`

Update a local copy of the PR: \
`$ git checkout pull/10937` \
`$ git pull https://git.openjdk.org/jdk pull/10937/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10937`

View PR using the GUI difftool: \
`$ git pr show -t 10937`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10937.diff">https://git.openjdk.org/jdk/pull/10937.diff</a>

</details>

### CODE REVIEW

**Comment by iklam** [comment]
Looks good. There's more to change:

- in cds.h, change `_space` to `_regions`, and increment`CURRENT_CDS_ARCHIVE_VERSION` by 1.
- in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/memory/FileMapInfo.java:
  -  `_space[0]` -> `_regions[0]`, 
  - `spaceField` -> `regions_0Field`
  - `space_0` -> `regions_0`, etc
  - This change is tested by test/hotspot/jtreg/serviceability/sa/ClhsdbCDSCore.java
- You will also see a bunch of C++ compilation errors to tell you what else need to be changed.

File: src/hotspot/share/cds/filemap.cpp [hunk 1]
```diff
@@ -1760,7 +1760,7 @@ size_t FileMapInfo::write_heap_regions(GrowableArray<MemRegion>* regions,
     if (size > 0) {
       int oopmap_idx = i * 2;
       int ptrmap_idx = i * 2 + 1;
-      space_at(region_idx)->init_bitmaps(bitmaps->at(oopmap_idx),
+      region_at(region_idx)->init_bitmaps(bitmaps->at(oopmap_idx),
```
Comment (iklam) [inline src/hotspot/share/cds/filemap.cpp:1763]: The next line need to be aligned.

**Review by iklam — APPROVED** [review]
Also the functions first_core_space/last_core_space in filemap.hpp need to be renamed. Please also check filemap.cpp to see if there are any other use of "space" that refers to a FileMapRegion.

**Review by iklam — CHANGES_REQUESTED** [review]

**Review by calvinccheung — APPROVED** [review]
Looks good.

One nit: please update copyright year 2021 to 2022 in cdsConstants.cpp.

**Review by iklam — APPROVED** [review]

**Pull request closed without merging**